decidim-budgets/app/commands/decidim/budgets/add_line_item.rb

```diff
@@ -10,7 +10,7 @@ class AddLineItem < Rectify::Command
       # project - The the project to include in the order
       # current_user - The current user logged in
       def initialize(current_order, project, current_user)
-        @order = current_order
+        @order = current_order.persisted? ? current_order : nil
```

Comment:
Sorry, but I can't find out why this change is needed for this feature. Is this fixing a new bug found?

Reply:
Yes found a bug and needed it 😉.
Bug: When €0 assigned it didn't show the threshold nor the minimum amount the participant required to vote. [screenshot €0 assigned](https://user-images.githubusercontent.com/210216/77104851-f7f6da00-6a1c-11ea-833f-79330837173b.png) and [screenshot with € assigned](https://user-images.githubusercontent.com/210216/77104995-34c2d100-6a1d-11ea-805d-edb2168780c7.png).

Before (this PR) the `current_order` could be nil and was created in the commands private method `:order` when the first project was added.

The order is needed before to know the rules to show to the participant, so now the `current_order` always returns an order object.

```rb
   @current_order ||= Order.find_by || Order.new
```

So this way the command works, and fixes the issue. 😁

https://github.com/decidim/decidim/blob/fbabd9e3513d38617325d901ba66695443a02514/decidim-budgets/app/controllers/concerns/decidim/budgets/needs_current_order.rb#L17-L19

Reply:
Sorry, I didn't see your question before. I think that is solved, but I can think in some improvements on it:
* You can use `find_or_initialize_by` instead of `find_by || new`, in this case I think that it would do the exactly the same that you are doing
* I wouldn't add the `persisted?` related logic to the command. I'd add a `persisted_current_order` getter to the controller and I'd use it when calling the command.

Does this help you? 😬

Reply:
thanks @leio10 , modified as you suggested. 
Yes, Certainly `find_or_initialize_by` is for this use cases.